pkg/docker/docker_client.go

```diff
@@ -132,6 +146,69 @@
 	return dockerClient, dockerHostInRemote, err
 }
 
+// If the DOCKER_TLS_VERIFY environment variable is set
+// this function returns HTTP client with appropriately configured TLS config.
+// Otherwise, nil is returned.
+func newHttpClient() *http.Client {
+	tlsVerifyStr, tlsVerifyChanged := os.LookupEnv("DOCKER_TLS_VERIFY")
+
+	if !tlsVerifyChanged {
+		return nil
+	}
+
+	var tlsOpts []func(*tls.Config)
+
+	tlsVerify, _ := strconv.ParseBool(tlsVerifyStr)
+	if !tlsVerify {
+		tlsOpts = append(tlsOpts, func(t *tls.Config) {
+			t.InsecureSkipVerify = true
```

Comment:
@matejvasek Will check it out, thanks!

Reply:
+1

Reply:
Bud `docker` CLI does this too.

Reply:
Without this `DOCKER_TLS_VERIFY=0` wouldn't have desired effect, right?

Reply:
@lance @evankanderson wdyt?

Reply:
I could however improve the condition. Right now `DOCKER_TLS_VERIFY="some-non-bool"` causes that verification is skipped. It should be the other way around: if the value is not bool it should be considered to be true.

Reply:
I've just committed fixup.

Reply:
It's my understanding with this line, the certificate chain isn't checked which kind of defeats the whole point right? When verify is 0, it should fall back to unencrypted.

Reply:
`DOCKER_TLS_VERIFY=0` does uses end to end encryption:
data are still encrypted, nobody can see what you are sending,
however it does not verify certificate of server so there may be impostor.
This is what original `docker` CLI does.

Reply:
(When I set verify to 0 on Windows, it falls back to an unencrypted channel, not an unvalidated encrypted channel.)

Reply:
So `DOCKER_TLS_VERIFY=0` is less secure than `DOCKER_TLS_VERIFY=1`, but it's still more secure that if the envvar is not set at all.

Reply:
That's how I understand it.

Reply:
> (When I set verify to 0 on Windows, it falls back to an unencrypted channel, not an unvalidated encrypted channel.)

Interesting with both it works for me on Linux. In both cases it can talk to server using https not http.

Reply:
```
docker image ls | wc -l
error during connect: Get "http://localhost:2376/v1.24/images/json": read tcp 127.0.0.1:54338->127.0.0.1:2376: read: connection reset by peer
0

DOCKER_TLS_VERIFY=1 docker image ls | wc -l
670

DOCKER_TLS_VERIFY=0 docker image ls | wc -l
670
```

Reply:
But I am using `socat` as https endpoint since my docker daemon runs on unix socket.

Reply:
Hmm let me check, maybe I am doing something wrong with `socat` proxying.

Reply:
Happy to defer to you on this since you're knee deep in Docker CLI code. 👍

Reply:
OK I can reproduce it, my bad!

Reply:
In my opinion.

Reply:
also:
https://github.com/docker/cli/blob/82427d1a078e89d9bd10dd397e5ee7736a8086f9/cli/context/docker/load.go#L82-L86

Reply:
@riverar you are right about behaviour of the envvar -- it only matter if its set/empty.
The effect that I described (encryption+non_verify_cert) can be achieved by `--tlsverify=false` flag.

So I immodestly dare to say that my implementation here is little bit better.
The thing is that we do not have `--tlsverify` flag so to achieve similar effect we can use the envvar even if it is not identical to what `docker` CLI does.

Do you agree.

Reply:
Agreed. (I continue to be extremely disappointed with Docker, esp. its impl. on Windows.)

Reply:
@matejvasek Ah ha! It does indeed. (That's miserable behavior, imo.)